beatrix/src/test/java/org/killbill/billing/beatrix/integration/TestCatalogRetireElements.java

```diff
@@ -107,10 +109,17 @@ public void testChangePlanTwiceWithNewPlan() throws Exception {
         // The code normally goes through the grandfathering logic to find the version but specifies the transitionTime of the latest CHANGE (and not the subscriptionStartDate)
         // and therefore correctly find the latest catalog version, invoicing at the new price 295.95
         //
-        invoiceChecker.checkInvoice(account.getId(), 4, callContext,
+        Invoice curInvoice = invoiceChecker.checkInvoice(account.getId(), 4, callContext,
                                     new ExpectedInvoiceItemCheck(new LocalDate(2015, 12, 5), new LocalDate(2016, 1, 5), InvoiceItemType.RECURRING, new BigDecimal("295.95")),
                                     new ExpectedInvoiceItemCheck(new LocalDate(2015, 12, 5), new LocalDate(2016, 1, 5), InvoiceItemType.REPAIR_ADJ, new BigDecimal("-500.00")),
                                     new ExpectedInvoiceItemCheck(new LocalDate(2015, 12, 5), new LocalDate(2015, 12, 5), InvoiceItemType.CBA_ADJ, new BigDecimal("204.05")));
+        final VersionedCatalog catalog = catalogUserApi.getCatalog("foo", callContext);
+        // RECURRING should be set against V2
+        Assert.assertEquals(curInvoice.getInvoiceItems().get(0).getCatalogEffectiveDate(), catalog.getVersions().get(1).getEffectiveDate());
```

Comment:
I don't know about easier because now every test would need to pull the catalog, and understand which catalog version to pass, but i agree this would make the testing more systematic, probably a good thing from a regression point of view - we can talk to decide whether this is worth the effort.